core/document/src/lib.rs

```diff
@@ -22,31 +26,148 @@ impl SvgElement {
 	}
 }
 
-#[derive(Default, Debug, Clone, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub enum DocumentError {
+	ElementNotFound,
+	NotAShape,
+	ElementAlreadyExists,
+	InvalidPath,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Folder {
+	elements: HashMap<String, SvgElement>,
```

Comment:
I'm thinking we want to use a vector not a hashmap. Names should be a property on every `SvgElement`, not something intrinsic (and strictly unique) to folders. Most shapes won't be named, unless you go through the effort to rename them, and in that case there should be not requirement about name uniqueness. Using a vector instead would also help avoid the requirement of naming folders arbitrarily, like where you concatenate in the `fold()` function above.